bin/tape

```diff
@@ -41,18 +46,31 @@ if (typeof opts.ignore === 'string') {
     var matcher = ignore.createMatcher(ignoreStr);
 }
 
-opts._.forEach(function (arg) {
-    // If glob does not match, `files` will be an empty array.
-    // Note: `glob.sync` may throw an error and crash the node process.
-    var files = glob.sync(arg);
+hasDynamicImport().then(function (hasSupport) {
+    var importOrRequire = hasSupport ? require('./import-or-require') : function (file) {
+        require(file);
+    };
 
-    if (!Array.isArray(files)) {
-        throw new TypeError('unknown error: glob.sync did not return an array or throw. Please report this.');
-    }
+    var loadingPromise = opts._.reduce(function (loadingPromise, arg) {
+        // If glob does not match, `files` will be an empty array.
+        // Note: `glob.sync` may throw an error and crash the node process.
+        var files = glob.sync(arg);
+
+        if (!Array.isArray(files)) {
+            throw new TypeError('unknown error: glob.sync did not return an array or throw. Please report this.');
```

Comment:
this will now reject the Promise, instead of exiting the program

Reply:
i moved the glob function into a reducer outside the promise. i also updated the error message to read

```js
throw new TypeError('unknown error: glob.sync("' + arg + '") did not return an array or throw. Please report this.');
```

to make it easier to see which glob call failed.